altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -87,25 +89,24 @@ pub fn switch_context() {
 ///Get the next task from the queue
 fn schedule_next_task() -> bool {
   let mut queues = Priority::all();
-  unsafe {
-    if CRITICAL_TASK_COUNTER >= CRITICAL_TASK_MAX {
-      queues = 1..NUM_PRIORITIES;
-      CRITICAL_TASK_COUNTER = 0;
-    }
+  if CRITICAL_TASK_COUNTER.load(Ordering::Relaxed)  >= CRITICAL_TASK_MAX {
+    queues = 2..NUM_PRIORITIES;
```

Comment:
This will still skip over `Critical` tasks if the counter reaches the max, not just `Normal`. Consider adding a function on the `Priority` type to exclude a certain level, `Priority::all_except(priority: Priority)` perhaps?